### PR TITLE
[Chore]: Technical: Translate utils (color and data)

### PR DESCRIPTION
**Pull request by HeimEndyd**

### CODE REVIEW

File: src/utils/color-utils.ts [hunk 1]
```diff
@@ -38,7 +41,7 @@ export function hexToRgb(hex) {
   return [r, g, b];
 }
 
-export function isHexColor(hex) {
+export function isHexColor(hex: string) {
```
Comment (heshan0131) [inline src/utils/color-utils.ts:44]: Use type guard here `isHexColor(hex:string): hex is HexColor`

File: src/utils/data-utils.ts [hunk 2]
```diff
@@ -336,11 +343,10 @@ export const arrayMove = (array, from, to) => {
 /**
  * Get the value format based on field and format options
  * Used in render tooltip value
- * @type {typeof import('./data-utils').getFormatter}
  * @param format
  * @param field
  */
-export function getFormatter(format, field) {
+export function getFormatter(format: any, field?: Field): FieldFormatter {
```
Comment (heshan0131) [inline src/utils/data-utils.ts:349]: `format: string | Record<string, string>`

**Review by heshan0131 — APPROVED** [review]

File: src/utils/color-utils.ts (hunk 1 above)
Comment (HeimEndyd) [reply src/utils/color-utils.ts]: this function works differently, it returns a RegExpExecArray used to get the RGB component of the hex color. 
Should I refactor this?

Comment (heshan0131) [reply src/utils/color-utils.ts:44]: oh okay, nvm. disregard my comment. this should `isHexColor(hex: string): RGBColor | null`

**Review by heshan0131 — APPROVED** [review]

**Pull request merged**